libcodechecker/server/api/config_handler.py

```diff
@@ -0,0 +1,83 @@
+# --------------------------------------------------------------------------
+#                     The CodeChecker Infrastructure
+#   This file is distributed under the University of Illinois Open Source
+#   License. See LICENSE.TXT for details.
+# -------------------------------------------------------------------------
+"""
+Handle Thrift requests for confugyration.
+"""
+from Configuration_v6.ttypes import *
+
+from libcodechecker.logger import get_logger
+from libcodechecker.profiler import timeit
+
+from libcodechecker.server import permissions
+from libcodechecker.server.database.config_db_model import Configuration
+from libcodechecker.util import DBSession
+
+LOG = get_logger('server')
+
+
+class ThriftConfigHandler(object):
+    """
+    Manages Thrift requests regarding configuration.
+    """
+
+    def __init__(self, auth_session, config_session):
+        self.__auth_session = auth_session
+        self.__session = config_session
+
+    def __require_supermission(self):
+        """
+        Checks if the current user isn't a SUPERUSER.
+        """
+
+        try:
+            session = self.__session()
+            if not (permissions.require_permission(
+                    permissions.SUPERUSER,
+                    {'config_db_session': session},
+                    self.__auth_session)):
+                raise shared.ttypes.RequestFailed(
+                    shared.ttypes.ErrorCode.UNAUTHORIZED,
+                    "You are not authorized to execute this action.")
+
+            return True
+
+        except sqlalchemy.exc.SQLAlchemyError as alchemy_ex:
+            msg = str(alchemy_ex)
+            LOG.error(msg)
+            raise shared.ttypes.RequestFailed(shared.ttypes.ErrorCode.DATABASE,
+                                              msg)
+        finally:
+            if session:
+                session.close()
+
+    @timeit
+    def getNotification(self):
+        """
+        Retrieves the notification banner text.
+        """
+
+        notificationString = ''
+        with DBSession(self.__session) as configuration:
+            notificationQuery = configuration.query(Configuration).one()
+
+            notificationString = notificationQuery.notification
+        return notificationString
+
+    @timeit
+    def setNotification(self, newNotification):
+        """
+        Sets the notification banner remove_products_except.
+        Bevare: This method only works if the use is a SUPERUSER.
+        """
+
+        self.__require_supermission()
+
+        with DBSession(self.__session) as configuration:
+            notificationQuery = configuration.query(Configuration).one()
```

Comment:
If we hadn't set any notification text before it will return `None` and later you will try to set the `notification` member of this which will throw an error.